glusterd2/bricksplanner/utils.go

```diff
@@ -106,3 +107,34 @@ func getAvailableVgs(req *api.VolCreateReq) ([]Vg, error) {
 
 	return vgs, nil
 }
+
+// GetNewBrick creates new brick.
+func GetNewBrick(availableVgs []Vg, brickInfo brick.Brickstatus, vol *volume.Volinfo, brickCount int) api.BrickReq {
+	var newBrick api.BrickReq
+	for _, vg := range availableVgs {
+		if vg.AvailableSize >= brickInfo.Size.Capacity {
+			brickSize := brickInfo.Size.Capacity
+			lvName := fmt.Sprintf("brick_%s_s%d_b%d", vol.Name, brickCount, brickCount)
+			brickTpSize := uint64(float64(brickSize) * vol.SnapshotReserveFactor)
+			newBrick = api.BrickReq{
+				Type:           "Brick",
+				Path:           brickInfo.Info.Path,
+				BrickDirSuffix: "/brick",
+				TpName:         fmt.Sprintf("tp_%s_s%d_b%d", vol.Name, brickCount, brickCount),
+				LvName:         lvName,
+				Size:           brickSize,
+				TpSize:         brickTpSize,
+				TpMetadataSize: deviceutils.GetPoolMetadataSize(brickTpSize),
+				FsType:         "xfs",
+				MntOpts:        "rw,inode64,noatime,nouuid",
+				PeerID:         vg.PeerID,
+				VgName:         vg.Name,
+				DevicePath:     "/dev/" + vg.Name + "/" + lvName,
+			}
+			vg.AvailableSize -= brickSize
```

Comment:
Reducing the size not required here beacause of break after this

Reply:
reducing vg size only required if in the same iteration selecting more than one brick. Actual vg size will be updated in transaction anyways.